soc/xtensa/esp32s2/CMakeLists.txt

```diff
@@ -3,4 +3,81 @@
 zephyr_sources(
   soc.c
   soc_cache.c
+  loader.c
   )
+
+# get code-partition slot0 address
+dt_nodelabel(dts_partition_path NODELABEL "slot0_partition")
+dt_reg_addr(img_0_off PATH ${dts_partition_path})
+
+# get code-partition boot address
+dt_nodelabel(dts_partition_path NODELABEL "boot_partition")
+dt_reg_addr(boot_off PATH ${dts_partition_path})
+
+# get flash size to use in esptool as string
+math(EXPR flashsize "${CONFIG_FLASH_SIZE} / 0x100000")
+string(CONCAT esptoolpy_flashsize ${flashsize} "MB")
```

Comment:
@tejlmand How does that sound? FYI, esptool requires "MB" snippet.

Reply:
You can just do:
```
--offset 0x8000 --flash-size ${esptoolpy_flashsize}MB ${ESP_IDF_PATH}/components/partition_table
```

below, and save the concatenation 😉